### PR TITLE
i18n: Chat

### PR DESCRIPTION
**Pull request by Stefanacef**
Chat is ready for localisation.

Demo:
- https://wedance-stefanacef.netlify.app/chat/ 
- https://wedance-stefanacef.netlify.app/chat/Stefana

Closes #105

### CODE REVIEW

File: locales/en.yml [hunk 1]
```diff
@@ -32,24 +32,27 @@ chat:
   title: Chat
   welcome: Start conversation with something nice.
   with: Chat with {username}
-  dateDiff: "{diff} ago"
+  dateDiff: '{diff} ago'
```
Comment (best-lucky1030) [inline locales/en.yml:35]: where is `{diff}`  from?
and that value is always changed or fixed?

Comment (Stefanacef) [reply locales/en.yml]: {diff} it was from an old version of the main code, I have refactor the code now

File: locales/en.yml [hunk 2]
```diff
@@ -28,28 +28,39 @@ myprofile:
   posts:
     add: Write a post
   title: My Profile
-chat:
+  chat:
```
Comment (razbakov) [inline locales/en.yml:31]: this change puts chat under `myprofile.chat`
```suggestion
chat:
```

File: locales/en.yml [hunk 3]
```diff
@@ -28,28 +28,39 @@ myprofile:
...
   title: Chat
-  welcome: Start conversation with something nice.
-  with: Chat with {username}
-  dateDiff: "{diff} ago"
+  conversation: 
+      new: Start conversations and they will appear here.
```
Comment (razbakov) [inline locales/en.yml:34]: identation problem. should be one space less in the beginning
```suggestion
     new: Start conversations and they will appear here.
```

File: locales/en.yml [hunk 4]
```diff
@@ -28,28 +28,39 @@ myprofile:
...
+     welcome: Start conversation with something nice.
```
Comment (razbakov) [inline locales/en.yml:35]: I would call it `empty` like on other pages

```suggestion
     empty: Start conversation with something nice.
```

File: locales/en.yml [hunk 5]
```diff
@@ -28,28 +28,39 @@ myprofile:
...
+  message:
+      new: Write a message
+      writer: by you
```
Comment (razbakov) [inline locales/en.yml:38]: I would call it `byYou` or `own` or something closer to meaning. It's not writer ;)

```suggestion
      byYou: by you
```

File: locales/en.yml [hunk 6]
```diff
@@ -28,28 +28,39 @@ myprofile:
...
+      submit: Send
+      unread:  haven't opened this chat yet
+      receiver: Chat with {receiver.username}
+ lastsaw: opened chat last time
```
Comment (razbakov) [inline locales/en.yml:42]: If there are already variables with some name, just re-use them. Also please use camelCase. Also there is again indentation problem.

```suggestion
lastSeen: opened chat last time
```

File: pages/chat/_slug.vue [hunk 7]
```diff
@@ -18,7 +18,7 @@
           <div class="text-xs space-x-1 text-gray-900 flex">
             <TAvatar name :uid="item.createdBy" />
             <span>•</span>
-            <div>{{ dateDiff(item.createdAt) }} ago</div>
+            <div>{{ dateDiff(item.createdAt) }} {{ $t('date') }}</div>
```
Comment (razbakov) [inline pages/chat/_slug.vue:21]: Please remove it. `dateDiff` function already includes `ago` word.
And also if you would use it `date` should belong to some group. It's too vague like that.

```suggestion
            <div>{{ dateDiff(item.createdAt) }}</div>
```

File: pages/chat/_slug.vue [hunk 8]
```diff
@@ -1,9 +1,9 @@
 <template>
   <TLoader v-if="loading" />
   <div v-else class="flex flex-col h-full">
-    <THeader :title="`Chat with ${receiver.username}`" />
+    <THeader :title="$t(`chat.message.receiver, ${receiver.username}`)" />
```
Comment (razbakov) [inline pages/chat/_slug.vue:4]: That won't work. Here is the right way:

```suggestion
    <THeader :title="$t('chat.message.receiver', { username: receiver.username })" />
```

File: locales/en.yml [hunk 9]
```diff
@@ -28,28 +28,39 @@ myprofile:
...
-  dateDiff: "{diff} ago"
+  conversation: 
+      new: Start conversations and they will appear here.
+     welcome: Start conversation with something nice.
+  message:
+      new: Write a message
+      writer: by you
+      submit: Send
+      unread:  haven't opened this chat yet
+      receiver: Chat with {receiver.username}
```
Comment (razbakov) [inline locales/en.yml:41]:
```suggestion
      receiver: Chat with {username}
```

File: pages/chat/_slug.vue (hunk 8 above)
Comment (razbakov) [reply pages/chat/_slug.vue:4]: What's the difference between chat, conversation and message?

File: pages/chat/index.vue [hunk 10]
```diff
@@ -1,8 +1,8 @@
 <template>
   <div>
-    <THeader title="Chat" />
+    <THeader :title="$t('chat.title')" />
     <div v-if="!chats" class="p-16 text-xs text-center">
-      Start conversations and they will appear here.
+      {{ $t('chat.conversation.new') }}
```
Comment (razbakov) [inline pages/chat/index.vue:5]: I would call this one also `empty`

```suggestion
      {{ $t('chat.empty') }}
```

File: pages/chat/index.vue [hunk 11]
```diff
@@ -20,11 +20,12 @@
             <span>{{ chat.receiver.username || 'DELETED' }}</span>
```
Comment (razbakov) [inline pages/chat/index.vue:20]: One string missed ;)

**Review by razbakov — CHANGES_REQUESTED** [review]

File: locales/en.yml (hunk 1 above)
Comment (razbakov) [reply locales/en.yml]: @best-lucky1030 read about i18n formatting here - https://kazupon.github.io/vue-i18n/guide/formatting.html
That's how you can insert dynamic variables into translation. We talk more about it in our daily meetings.

File: pages/chat/index.vue [hunk 12]
```diff
@@ -17,14 +17,15 @@
         </div>
         <div class="flex-grow">
           <div class="text-xs space-x-1 text-gray-900 flex">
-            <span>{{ chat.receiver.username || 'DELETED' }}</span>
+            <span>{{ chat.receiver.username || $t('chat.delete') }}</span>
```
Comment (razbakov) [inline pages/chat/index.vue:20]:
```suggestion
            <span>{{ chat.receiver.username || $t('chat.receiver.deleted') }}</span>
```

**Review by razbakov — CHANGES_REQUESTED** [review]

**Review by razbakov — APPROVED** [review]

**Pull request merged**